code/modules/mob/living/silicon/silicon.dm

```diff
@@ -51,8 +52,11 @@
 
 	var/obj/item/modular_computer/pda/silicon/modularInterface
 
+
 /mob/living/silicon/Initialize(mapload)
 	. = ..()
+	if(SStts.tts_enabled)
+		voice = pick(SStts.available_speakers)
```

Comment:
what happens if this runs before sstts init?

Reply:
it doesn't since this runs before SSatoms

Reply:
what about INITIALIZE_IMMEDIATE?

Reply:
in those cases, they'll just have to not add a voice. But in those cases, the person adding the voice should be aware of the immediate initialize so they'll have their own countermeasures.

Reply:
This is one of the reasons I get the feeling an event based system would be more robust

Reply:
It wouldn't work easily like that because you may want a specific voice instead of just picking a random voice. I could definitely look into implementing a feature that lets you give a random voice to an /atom/movable in a way that has little chance of failing, but I think it's not all that necessary to do right now.